AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginLazyLoadTests/LL12/CompositePK/GraphQLLazyLoadCompositePKChildTests.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Combine
+import XCTest
+
+@testable import Amplify
+import AWSPluginsCore
+
+extension GraphQLLazyLoadCompositePKTests {
+
+    // MARK: - CompositePKParent / CompositePKChild
+    
+    func initChild(with parent: CompositePKParent? = nil) -> CompositePKChild {
+        CompositePKChild(childId: UUID().uuidString, content: "content", parent: parent)
+    }
+    
+    func testSaveCompositePKChild() async throws {
+        await setup(withModels: CompositePKModels())
+        
+        let parent = initParent()
+        let savedParent = try await mutate(.create(parent))
+        let child = initChild(with: savedParent)
+        try await mutate(.create(child))
+    }
+    
+    func testUpdateCompositePKChild() async throws {
+        await setup(withModels: CompositePKModels())
+        let parent = initParent()
+        let savedParent = try await mutate(.create(parent))
+        let child = initChild(with: parent)
+        var savedChild = try await mutate(.create(child))
+        let loadedParent = try await savedChild.parent
+        XCTAssertEqual(loadedParent?.identifier, savedParent.identifier)
+        
+        // update the child to a new parent
+        let newParent = initParent()
+        let savedNewParent = try await mutate(.create(newParent))
+        savedChild.setParent(savedNewParent)
+        let updatedChild = try await mutate(.update(savedChild))
+        let loadedNewParent = try await updatedChild.parent
+        XCTAssertEqual(loadedNewParent?.identifier, savedNewParent.identifier)
+    }
+    
+    func testUpdateFromNoParentCompositePKChild() async throws {
+        await setup(withModels: CompositePKModels())
+        let parent = initParent()
+        let savedParent = try await mutate(.create(parent))
+        
+        let childWithoutParent = initChild()
+        var savedChild = try await mutate(.create(childWithoutParent))
+        let nilParent = try await savedChild.parent
+        XCTAssertNil(nilParent)
```

Comment:
This test was what triggered the implementation change in the PR, the child without a parent should be query-able and can be traversed to the parent, even if the parent is nil. Previously, this call was throwing due to the default model provider throwing. Since the parent is an optional association, LazyReference gets `nil` back from the default model provider, and returns `nil` on `get()`.